NDTensors/src/dense/generic_array_constructors.jl

```diff
@@ -0,0 +1,31 @@
+using .TypeParameterAccessors:
+  default_type_parameter,
+  parenttype,
+  set_eltype,
+  specify_default_type_parameters,
+  type_parameter
+##TODO replace randn in ITensors with generic_randn
+## and replace zeros with generic_zeros
+
+# This is a file to write generic fills for NDTensors.
+#  This includes random fills, zeros, ...
+
+function generic_randn(StoreT::Type{<:Dense}, dims::Integer; rng=Random.default_rng())
+  StoreT = specify_default_type_parameters(StoreT)
+  DataT = specify_type_parameter(type_parameter(StoreT, parenttype), eltype, eltype(StoreT))
+  @assert eltype(StoreT) == eltype(DataT)
+
+  data = generic_randn(DataT, dims; rng=rng)
+  StoreT = set_datatype(StoreT, typeof(data))
+  return StoreT(data)
+end
+
+function generic_zeros(StoreT::Type{<:Dense}, dims::Tuple{Integer})
```

Comment:
```suggestion
function generic_zeros(StoreT::Type{<:Dense}, dims::Integer)
```
to be consistent with `generic_randn`.

Reply:
`generic_zeros(StoreT::Type{<:AbstractArray}, dims...)` definitions should be updated accordingly.

Reply:
In order to have both of these lines work
```
generic_randn(Dense, 2)
generic_randn(Dense, (2,))
``` 
I updated the `generic_randn` definition to `generic_randn(StoreT::Type{<:Dense}, dims::Tuple{Integer})`. Previously this would work because the tuple would be piped to an int with the `generic_randn(arraytype, dims...)` call but now the Tuple version goes into the `AbstractArray` version of the function.

I don't understand what you mean by 
```
generic_zeros(StoreT::Type{<:AbstractArray}, dims...) definitions should be updated accordingly.
```

Reply:
I think we are talking past each other here, why don't you do your best to implement things as I've described and let me know when it is ready to be reviewed.